modules/nlu/src/backend/training-pipeline.ts

```diff
@@ -26,9 +25,12 @@ import {
   TrainingSession
 } from './typings'
 import Utterance, { buildUtteranceBatch, UtteranceToken, UtteranceToStringOptions } from './utterance/utterance'
-
-// TODO make this return artefacts only and move the make model login in E2
-export type Trainer = (input: TrainInput, tools: Tools) => Promise<Model>
+export class TrainingCanceledError extends Error {
```

Comment:
There's no need for this to be moved at the top of this file, but honnestly, I don't care.

Just make sure, in future PR's you don't make changes that don't have to be. It makes more code to review.